src/builder.rs

```diff
@@ -9,6 +9,24 @@ use crate::transport::TransportsError;
 use crate::{DeviceId, DistinctId, Map, system_snapshot::SystemSnapshotter};
 use crate::{Groups, Recorder, Worker};
 
+macro_rules! build_with_default_storage {
```

Comment:
This is a bit weird, but Rust doesn't like that I'm constructing two different impls of the same trait ... instead of doing some wacky enum stuff, I figure we can just macro the pain away.